### PR TITLE
factor notehead offset into stem length calculations

### PR DESCRIPTION
**Pull request by miiizen**
Resolves: #19488 
Stems no longer extend when changing notehead offset.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

**Comment by zacjansheski** [comment]
Stems still extend when they are beamed

https://github.com/musescore/MuseScore/assets/69917893/ce1786f4-70db-43b3-96d4-f3cf876aa394

File: src/engraving/dom/chord.cpp [hunk 1]
```diff
@@ -1420,7 +1420,7 @@ double Chord::calcDefaultStemLength()
         // when the chord's magnitude is < 1, the stem length with mag can find itself below the middle line.
         // in those cases, we have to add the extra amount to it to bring it to a minimum.
         double upValue = m_up ? -1. : 1.;
-        double stemStart = startNote->pos().y();
+        double stemStart = startNote->pos().y() - startNote->offset().y();
```
Comment (cbjeukendrup) [inline src/engraving/dom/chord.cpp:1423]: Same as https://github.com/musescore/MuseScore/pull/19559/files#r1341666459; `startNote->layoutData()->pos()` would be a more direct way of getting the desired value

**Comment by miiizen** [comment]
I've prevented the beamed notes from extending by subtracting the notehead offset in the value returned by `BeamTremoloLayout::chordBeamAnchorY`.  This also prevents beams moving with the notehead offset.  However this is less consistent with `BeamTremoloLayout::chordBeamAnchorX`, as this includes offset in the value it returns.  Would subtracting the offset each time `BeamTremoloLayout::chordBeamAnchor` is called be preferred?

**Comment by zacjansheski** [comment]
> I've prevented the beamed notes stems from extending by subtracting the notehead offset in the value returned by `BeamTremoloLayout::chordBeamAnchorY`. This also prevents beams moving with the notehead offset. However this is less consistent with `BeamTremoloLayout::chordBeamAnchorX`, as this includes offset in the value it returns. Would subtracting the offset each time `BeamTremoloLayout::chordBeamAnchor` is called be preferred?

The current solution seems good to me.  Are there any anticipated problems related to the `BeamTremoloLayout::chordBeamAnchorX` inconsistency?

**Comment by cbjeukendrup** [comment]
I think it's fine; actually it's not that inconsistent, because if I'm not mistaken, the implementation of `chordBeamAnchorX` does ultimately not work with notes at all and thus doesn't take any note offset into account anyway.

**Review by cbjeukendrup — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
@zacjansheski Shall we merge it in that case? 
(@miiizen Any objections?)

**Comment by cbjeukendrup** [comment]
Oh, wait a minute, a rebase is needed as `layoutData` has been renamed to `ldata`!

**Comment by miiizen** [comment]
@cbjeukendrup no, as long as you're happy with it go ahead!

**Pull request merged**